x-pack/plugins/transform/server/lib/alerting/transform_health_rule_type/transform_health_service.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from 'kibana/server';
+import { i18n } from '@kbn/i18n';
+import type { Transform as EsTransform } from '@elastic/elasticsearch/api/types';
+import { TransformHealthRuleParams } from './schema';
+import {
+  ALL_TRANSFORMS_SELECTION,
+  TRANSFORM_HEALTH_CHECK_NAMES,
+} from '../../../../common/constants';
+import { getResultTestConfig } from '../../../../common/utils/alerts';
+import {
+  NotStartedTransformResponse,
+  TransformHealthAlertContext,
+} from './register_transform_health_rule_type';
+
+interface TestResult {
+  name: string;
+  context: TransformHealthAlertContext;
+}
+
+// @ts-ignore FIXME update types in the elasticsearch client
+type Transform = EsTransform & { id: string; description?: string; sync: object };
+
+export function transformHealthServiceProvider(esClient: ElasticsearchClient) {
+  const transformsDict = new Map<string, Transform>();
+
+  /**
+   * Resolves result transform selection.
+   * @param includeTransforms
+   * @param excludeTransforms
+   */
+  const getResultsTransformIds = async (
+    includeTransforms: string[],
+    excludeTransforms: string[] | null
+  ): Promise<string[]> => {
+    const includeAll = includeTransforms.some((id) => id === ALL_TRANSFORMS_SELECTION);
+
+    // Fetch transforms to make sure assigned transforms exists.
+    const transformsResponse = (
+      await esClient.transform.getTransform({
+        ...(includeAll ? {} : { transform_id: includeTransforms.join(',') }),
+        allow_no_match: true,
+      })
+    ).body.transforms as Transform[];
+
+    let resultTransformIds: string[] = [];
+
+    transformsResponse.forEach((t) => {
+      transformsDict.set(t.id, t);
+      if (t.sync) {
+        resultTransformIds.push(t.id);
+      }
+    });
+
+    if (excludeTransforms && excludeTransforms.length > 0) {
```

Comment:
Is the check for excluded transforms working? I configured a transform to exclude, but still receive an alert for it:

![image](https://user-images.githubusercontent.com/7405507/133600646-c43afc41-d4e4-472f-ae46-d51dbf2ecb7f.png)

![image](https://user-images.githubusercontent.com/7405507/133600697-e979c1d5-0008-4167-8439-8b6e57fb4c24.png)

Reply:
Here I seem to get notified about the transform(s) that I have excluded, and not the one(s) I have included.

Reply:
Fixed in https://github.com/elastic/kibana/pull/112277/commits/4a66cd50e3a14683934d87295255e9959d2b9f59